mgnify_pipelines_toolkit/analysis/shared/amrintegrator.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2024-2025 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import fileinput
+import logging
+import os
+from collections import defaultdict
+from typing import Dict, List, Optional, Union, DefaultDict
+
+# Constant to simplify long strings
+DRUG_REPLACEMENT_STRINGS: Dict[str, str] = {
+    "macrolide antibiotic; lincosamide antibiotic; streptogramin antibiotic": "MLS",
+    "lincosamide/macrolide/streptogramin": "MLS",
+}
+
+# Set up logger
+logger = logging.getLogger(__name__)
+
+
+def validate_inputs(optional_inputs: Dict[str, Optional[str]]) -> List[str]:
+    """
+    Validate that input files exist and return a list of valid input names.
+
+    Args:
+        optional_inputs: Dictionary mapping input names to file paths (can be None)
+
+    Returns:
+        List of valid input names whose files exist
+    """
+    valid_inputs: List[str] = []
+    for name, path in optional_inputs.items():
+        if path and os.path.exists(path):
+            valid_inputs.append(name)
+        elif path:
+            logger.warning(f"File not found for '{name}' → {path}")
+    return valid_inputs
+
+
+def normalize_drug_class(drug_class: str) -> str:
+    """
+    Normalize drug class names by applying replacements and removing suffixes.
+
+    Args:
+        drug_class: Raw drug class string
+
+    Returns:
+        Normalized drug class string
+    """
+    # Apply known full-string replacements
+    for original, replacement in DRUG_REPLACEMENT_STRINGS.items():
+        if original in drug_class:
+            drug_class = drug_class.replace(original, replacement)
+
+    # Remove generic suffixes after replacement
+    drug_class = drug_class.replace(" antibiotic", "")
+
+    return drug_class
+
+
+def parse_hamronized(
+    amr_annotation: DefaultDict[str, Dict[str, Dict[str, Union[List[str], str]]]], input_file: str
```

Comment:
Comment - something to improve later. If the type get this convoluted: `DefaultDict[str, Dict[str, Dict[str, Union[List[str], str]]]]` we should think about a different data structure to handle these (a pydantic model maybe). Anyway... this is pretty nice (progressive improvements :))